Modules/_hashopenssl.c

```diff
@@ -455,6 +456,13 @@ EVPnew(PyObject *name_obj,
         return NULL;
     }
 
+    /*
+     * Load config file, and other important initialisation in order to detect
+     * architecture optimizations correctly. E.g: POWER8 optimization
+     *
+     * Source: https://wiki.openssl.org/index.php/Libcrypto_API
+     */
+    OPENSSL_config(NULL);
```

Comment:
I would prefer to see this call in PyInit__hashlib() after ERR_load_crypto_strings().